src/components/MultiHopTrade/components/LimitOrder/components/CancelLimtOrder.tsx

```diff
@@ -0,0 +1,222 @@
+import {
+  Button,
+  Card,
+  Flex,
+  Heading,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Progress,
+  Stack,
+} from '@chakra-ui/react'
+import { SwapperName } from '@shapeshiftoss/swapper'
+import type { CowSwapError } from '@shapeshiftoss/types'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { bn, bnOrZero, fromBaseUnit } from '@shapeshiftoss/utils'
+import { formatDistanceToNow } from 'date-fns'
+import { useCallback, useEffect, useMemo } from 'react'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIconWithBadge } from 'components/AssetIconWithBadge'
+import { Row } from 'components/Row/Row'
+import { RawText, Text } from 'components/Text'
+import { TransactionTypeIcon } from 'components/TransactionHistory/TransactionTypeIcon'
+import { useActions } from 'hooks/useActions'
+import { useErrorHandler } from 'hooks/useErrorToast/useErrorToast'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { useCancelLimitOrdersMutation } from 'state/apis/limit-orders/limitOrderApi'
+import { limitOrderSlice } from 'state/slices/limitOrderSlice/limitOrderSlice'
+import { selectOrderToCancel } from 'state/slices/limitOrderSlice/selectors'
+import { selectAssetById, selectFeeAssetById } from 'state/slices/selectors'
+import { useAppSelector, useSelectorWithArgs } from 'state/store'
+
+import { SwapperIcon } from '../../TradeInput/components/SwapperIcon/SwapperIcon'
+
+const cardBorderRadius = { base: '2xl' }
+
+export const CancelLimitOrder = () => {
+  const wallet = useWallet().state.wallet
+  const { setOrderToCancel } = useActions(limitOrderSlice.actions)
+  const { showErrorToast } = useErrorHandler()
+
+  const orderToCancel = useAppSelector(selectOrderToCancel)
+
+  const [cancelLimitOrders, { data: wasCancellationSuccessful, error, isLoading, reset }] =
+    useCancelLimitOrdersMutation()
+
+  useEffect(() => {
+    if (!error) return
+
+    const description = (error as CowSwapError).description ?? 'Unknown Error'
+
+    // TODO: Actually render a translated error description.
+    showErrorToast(description)
+  }, [error, showErrorToast])
+
+  useEffect(() => {
+    // Must explicitly check for false, because undefined means a cancellation isn't completed.
+    if (wasCancellationSuccessful !== false) return
+
+    showErrorToast(
+      `Failed to cancel order uid ${orderToCancel?.order.uid}`,
+      'limitOrder.cancel.cancellationFailed',
+    )
+  }, [error, orderToCancel?.order.uid, showErrorToast, wasCancellationSuccessful])
```

Comment:
(unless we actually leverage `error`, see above)
```suggestion
  }, [orderToCancel?.order.uid, showErrorToast, wasCancellationSuccessful])
```

Reply:
2151427d82837a2c91febcaed514dcce0fdc662d